File: 158_two-strings-are-anagrams/two-strings-are-anagrams.cpp
/*
@Copyright:LintCode
@Author:   michael18
@Problem:  http://www.lintcode.com/problem/two-strings-are-anagrams
@Language: C++
@Datetime: 16-06-07 15:37
*/

#include <string.h>
class Solution {
public:
    /**
     * @param s: The first string
     * @param b: The second string
     * @return true or false
     */
    bool anagram(string s, string t) {
        // write your code here
        if(s.size() != t.size()){
            return false;
        }
       else {
            sort(s.begin(), s.end());
            sort(t.begin(), t.end());
            }
            return s==t;
        
    }
};

File: 158_two-strings-are-anagrams/two-strings-are-anagrams.java
/*
@Copyright:LintCode
@Author:   michael18
@Problem:  http://www.lintcode.com/problem/two-strings-are-anagrams
@Language: Java
@Datetime: 16-03-21 07:16
*/

public class Solution {
    /**
     * @param s: The first string
     * @param b: The second string
     * @return true or false
     */
    public boolean anagram(String s, String t) {
        // write your code here
        if (s == null || t == null){
            if (s == null && t == null){
                 return true; 
            }
            else{
                 return false;
            }
        }
        if (s.length() != t.length()){
            return false;
        }
        int[] ascii = new int[256];
        for (int i = 0; i < t.length(); i++){
            ascii[s.charAt(i)]++;
            ascii[t.charAt(i)]--;
        }
        for (int i = 0; i < 256; i++){
            if(ascii[i] != 0){
                return false;
            }
        }
        return true;
    }
}